src/pages/home/report/ReportActionCompose/ComposerWithSuggestions.js

```diff
@@ -0,0 +1,569 @@
+                const remainder = ComposerUtils.getCommonSuffixLength(commentRef.current, newComment);
+                setSelection({
+                    start: newComment.length - remainder,
+                    end: newComment.length - remainder,
+                });
+            }
+
+            // Indicate that draft has been created.
+            if (commentRef.current.length === 0 && newComment.length !== 0) {
+                Report.setReportWithDraft(reportID, true);
+            }
+
+            // The draft has been deleted.
+            if (newComment.length === 0) {
+                Report.setReportWithDraft(reportID, false);
+            }
+
+            commentRef.current = newComment;
+            if (shouldDebounceSaveComment) {
+                debouncedSaveReportComment(reportID, newComment);
+            } else {
+                Report.saveReportComment(reportID, newComment || '');
+            }
+            if (newComment) {
+                debouncedBroadcastUserIsTyping(reportID);
+            }
+        },
+        [debouncedUpdateFrequentlyUsedEmojis, preferredLocale, preferredSkinTone, reportID, setIsCommentEmpty],
+    );
+
+    /**
+     * Update the number of lines for a comment in Onyx
+     * @param {Number} numberOfLines
+     */
+    const updateNumberOfLines = useCallback(
+        (newNumberOfLines) => {
+            if (newNumberOfLines === numberOfLines) {
+                return;
+            }
+            Report.saveReportCommentNumberOfLines(reportID, newNumberOfLines);
+        },
+        [reportID, numberOfLines],
+    );
+
+    /**
+     * @returns {String}
+     */
+    const prepareCommentAndResetComposer = useCallback(() => {
+        const trimmedComment = commentRef.current.trim();
+        const commentLength = ReportUtils.getCommentLength(trimmedComment);
+
+        // Don't submit empty comments or comments that exceed the character limit
+        if (!commentLength || commentLength > CONST.MAX_COMMENT_LENGTH) {
+            return '';
+        }
+
+        updateComment('');
+        setTextInputShouldClear(true);
+        if (isComposerFullSize) {
+            Report.setIsComposerFullSize(reportID, false);
+        }
+        setIsFullComposerAvailable(false);
+        return trimmedComment;
+    }, [updateComment, setTextInputShouldClear, isComposerFullSize, setIsFullComposerAvailable, reportID]);
+
+    /**
+     * Callback to add whatever text is chosen into the main input (used f.e as callback for the emoji picker)
+     * @param {String} text
+     * @param {Boolean} shouldAddTrailSpace
+     */
+    const replaceSelectionWithText = useCallback(
+        (text, shouldAddTrailSpace = true) => {
+            const updatedText = shouldAddTrailSpace ? `${text} ` : text;
+            const selectionSpaceLength = shouldAddTrailSpace ? CONST.SPACE_LENGTH : 0;
+            updateComment(ComposerUtils.insertText(commentRef.current, selection, updatedText));
```

Comment:
This line caused this issue:
https://github.com/Expensify/App/issues/41778

More detail about the RCA here:
https://github.com/Expensify/App/issues/41778#issuecomment-2103018619